application/models/user_test.go

```diff
@@ -1079,109 +1079,146 @@ func (ms *ModelSuite) TestUser_WantsPostNotification() {
 	}
 }
 
-func (ms *ModelSuite) TestUser_GetPreferences() {
+func (ms *ModelSuite) TestUser_UpdateStandardPreferences() {
 	t := ms.T()
 
 	f := CreateUserFixtures_TestGetPreference(ms)
 
 	tests := []struct {
-		name string
-		user User
-		want []string // Preference Keys
+		name   string
+		user   User
+		SPrefs StandardPreferences
+		want   StandardPreferences
 	}{
-		{name: "has some", user: f.Users[0], want: []string{f.UserPreferences[0].Key, f.UserPreferences[1].Key}},
-		{name: "has none", user: f.Users[1], want: []string{}},
+		{
+			name: "Change Lang, Leave KG, Add TimeZone",
+			user: f.Users[0],
+			SPrefs: StandardPreferences{
+				Language:   domain.UserPreferenceLanguageFrench,
+				WeightUnit: domain.UserPreferenceWeightUnitKGs,
+				TimeZone:   "America/New York",
+			},
+			want: StandardPreferences{
+				Language:   domain.UserPreferenceLanguageFrench,
+				WeightUnit: domain.UserPreferenceWeightUnitKGs,
+				TimeZone:   "America/New York",
+			},
+		},
+		{
+			name: "Start with none then add one",
+			user: f.Users[1],
+			SPrefs: StandardPreferences{
+				Language: domain.UserPreferenceLanguageFrench,
+			},
+			want: StandardPreferences{
+				Language: domain.UserPreferenceLanguageFrench,
+			},
+		},
```

Comment:
Seems there could be other possibilities. For instance, will it work correctly if a user has one preference and another is added? Just to make certain that existing one doesn't get changed.

Reply:
That's what the previous test does.  The user has French and Kilograms and no Time Zone. The test changes French to English, leaves Kilograms as is and adds a Time Zone.

Is that what you mean?

Reply:
Yes, I think so. I admit I didn't study it that well.